lib/layout/data_grid.dart

```diff
@@ -356,7 +398,11 @@ class DataGridState extends WidgetState<DataGrid>
 
   void _buildDataRow() {
     _rows.clear();
-    for (Widget w in _children) {
+    // Added children.asMap().entries to have access for individual element index
+    for (final element in _children.asMap().entries) {
```

Comment:
I don't understand why you are converting a list into a map to get its index. Why not just do - 

for ( int index=0;index<_children.length;index++ ) {
  Widget w = _children[index];
...
}
that's all, why do you need a map for?